src/driver/driver.cpp

```diff
@@ -96,17 +96,25 @@ DriverStatus EvolutionDriver::Execute() {
     while (tm.KeepGoing() && signal != OutputSignal::analysis) {
       if (Globals::my_rank == 0) OutputCycleDiagnostics();
 
-      pmesh->PreStepUserWorkInLoop(pmesh, pinput, tm);
-      pmesh->PreStepUserDiagnosticsInLoop(pmesh, pinput, tm);
+      if (pmesh->PreStepUserWorkInLoop != nullptr) {
+        pmesh->PreStepUserWorkInLoop(pmesh, pinput, tm);
+      }
+      if (pmesh->PreStepUserDiagnosticsInLoop != nullptr) {
+        pmesh->PreStepUserDiagnosticsInLoop(pmesh, pinput, tm);
+      }
```

Comment:
Followup up from above: this is the change I was referring to, i.e., now needing to check for the function rather than just calling it.

Reply:
Oh right I see, yes I added checks that the function pointers are not `null` everywhere necessary in the code I think (if not I would expect the regression tests to break).